### ISSUE TITLE
Final Fantasy VIII (39150)

### ISSUE THREAD

**Issue by adrianlshaw** [ValveSoftware/Proton#696]
Edit: Game works very well with the exception that there is no music. Sound effects do work, however.

Specs: Proton 3.7-3, Ubuntu 18.04, NVIDIA Quadro FX 1800

**Comment by vandergail** [comment]
This game uses DirectMusic so the music is not working for me.

**Comment by Zero86Sk** [comment]
I can conform, music dosnt work.

**Comment by adrianlshaw** [comment]
Tested again and I noticed the same here. Music in the videos dies work but not in the game. Sound effects do work, however. I've updated the issue description to reflect this.

**Comment by nstgc** [comment]
Have any of you tried `WINEPREFIX=/path/to/prefix/ WINE=/path/to/proton/wine winetricks dmusic dsound`? Or perhaps `...winetricks  --force xact`?

**Comment by ryad-lindner** [comment]
@nstgc yes, I have tested `directmusic` (which includes `dsound`) and `xact`. Result: Unfortunately the game won't start, it hangs with a black screen and cracking sound. Only reverting to builtin libraries makes the game runnable again (of course without background music though).

**Comment by cdr255** [comment]
For me on Fedora 28, the advice [here](https://spcr.netlify.com/app/39150) to install directmusic via `WINEPREFIX="<Steam library path>/steamapps/compatdata/39150/pfx" winetricks directmusic` but then set dsound to builtin using `WINEPREFIX="<Steam library path>/steamapps/compatdata/39150/pfx" winecfg` made the stock background music play without issue.

**Comment by andreacorinti** [comment]
same here (Ubuntu 18.04)

I've also used this ->  https://steamcommunity.com/sharedfiles/filedetails/?id=594277140 because the original music from the PC version was really bad midi

**Comment by ryad-lindner** [comment]
I can confirm the workaround from @cdr255 works like a charm. Thank you very much!

**Comment by Pantar2205** [comment]
can confirm installing directmusic via winetricks makes game perfect. 37 hours played and can't find anything wrong.
1 error left is that when you quit the game the launcher for it crashes.
if music and that error is fixed, i would describe this title as flawless and it should be whitelisted.

thank you for your work

**Comment by jrobeson** [comment]
The game should run out of the box on Proton 4.2 without having to do anything like that.  Although there are some controller issues. I mentioned the workarounds here, for anybody having that problem. https://github.com/ValveSoftware/Proton/issues/1785#issuecomment-486944776

**Comment by shoober420** [comment]
# Compatibility Report
- Name of the game with compatibility issues: FINAL FANTASY VIII
- Steam AppID of the game: https://steamdb.info/app/39150/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.59
- Kernel version: 5.5.2
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 4.11-12

LOG:
[steam-39150.log](https://github.com/ValveSoftware/Proton/files/4176511/steam-39150.log)


## Symptoms
Game runs great, except theres no music as mentioned previously.

**Comment by landwill** [comment]
There have been a couple of reports about having save files lost / deleted; one person guessed that it may have been related to the Winetricks customization.
Has anything else been discovered since? For example, if very few users are affected by it, or that it's a one-time data loss that only occurs as you first apply the Winetricks customization.

**Comment by adrianlshaw** [comment]
I've had data loss on FF7 but I'm not sure whether it relates to Wine or to Square Enix's software.

**Comment by cybik** [comment]
It's possible that during the Opening Movie (remastered graphics mode active, fwiw), audio might hit a buffer underrun and start playing back music from a second before, resulting in an audio desync that may bother players expecting the intro movie's sound effects to match the actual movie (sword clangs, magic being cast, that sort of thing).

I'm also assuming this may hit on movie playback throughout the actual game, (again) fwiw.

I'm using GE 6.16 without any special protonfixes. (seems like I'm missing music)

**Comment by kisak-valve** [comment]
Any old game that uses DirectInput that features a launcher cannot be controlled with joystick

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7303.
@hfc2x posted on 2023-12-01T20:10:50:

# Compatibility Report
- Name of the game with compatibility issues:
Any older game that features a launcher that doesn't close upon launch. This particularly affects Final Fantasy games, like VII and VIII
- Steam AppID of the game: [39150](https://steamdb.info/app/39150/) (Final Fantasy VIII)

## System Information
- Kernel version: 6.6.2
- Proton version: Latest Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



## Symptoms 
A game such as Final Fantasy VIII does not get controller input with Steam Input because whenever it launches, it keeps 2 windows open, and Steam only sends inputs to the first window.

The launcher window stays in the background receiving inputs, and blocking the inputs from reaching the game, which only responds to keyboard key presses.

This not only affects Final Fantasy VIII, but this is the most notable example I can name from the top of my head.

This happens with X11. Haven't tried Wayland.

This does not happen in Windows because the launcher is seemingly able to pass down the inputs to the game.

## Reproduction

1. Launch Steam Input configuration for Final Fantasy VIII, and set the controller's buttons. Don't set them as keyboard key presses.
2. Launch the game from Steam and check the input options in the provided launcher.
3. Launcher responds to controller buttons.
4. Launch the game through launcher.
5. Game doesn't receive any inputs _but the launcher does_.
